packages/seda-chain-contracts/src/data_request_result.rs

```diff
@@ -1,52 +1,234 @@
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::{Deps, DepsMut, MessageInfo, Response, StdResult};
 
-use crate::msg::GetDataResultResponse;
-use crate::state::{DATA_REQUESTS_POOL, DATA_RESULTS};
+use crate::msg::{GetCommittedDataResultResponse, GetRevealedDataResultResponse};
+use crate::state::DATA_REQUESTS;
 use crate::types::Hash;
 
-use crate::state::DataResult;
 use crate::ContractError;
 
 pub mod data_request_results {
 
-    use crate::utils::check_eligibility;
+    use cosmwasm_std::{Addr, Env};
+    // use hex_literal::hex;
+    use sha3::{Digest, Keccak256};
+
+    use crate::{
+        msg::{
+            GetCommittedDataResultsResponse, GetCommittedExecutorsResponse, GetIdsResponse,
+            GetResolvedDataResultResponse, GetRevealedDataResultsResponse,
+        },
+        state::{DataResult, Reveal, DATA_RESULTS},
+        types::Bytes,
+        utils::check_eligibility,
+    };
 
     use super::*;
 
-    /// Posts a data result of a data request with an attached result.
+    /// Posts a data result of a data request with an attached hash of the answer and salt.
     /// This removes the data request from the pool and creates a new entry in the data results.
-    pub fn post_data_result(
+    pub fn commit_result(
         deps: DepsMut,
         info: MessageInfo,
         dr_id: Hash,
-        result: String,
+        commitment: Hash,
     ) -> Result<Response, ContractError> {
-        assert!(check_eligibility(&deps, info.sender)?);
+        assert!(check_eligibility(&deps, info.sender.clone())?);
         // find the data request from the pool (if it exists, otherwise error)
-        let dr = DATA_REQUESTS_POOL.load(deps.storage, dr_id.clone())?;
-        let dr_result = DataResult {
-            dr_id: dr.dr_id,
-            nonce: dr.nonce,
-            value: dr.value,
-            result: result.clone(),
-            chain_id: dr.chain_id,
-        };
+        let mut dr = DATA_REQUESTS.load(deps.storage, dr_id.clone())?;
```

Comment:
The load function may also be able to take a [reference](https://docs.rs/cw-storage-plus/latest/cw_storage_plus/struct.Map.html#method.load) for the second arg instead of having to clone.

Reply:
Same here, unfortunately, it doesn't accept a reference.